packages/lesswrong/components/users/DialogueMatchingPage.tsx

```diff
@@ -480,56 +805,77 @@ const DialogueCheckBox: React.FC<{
   );
 };
 
-const isMatched = (userDialogueChecks: DialogueCheckInfo[], targetUserId: string): boolean => {
-  return userDialogueChecks.some(check => check.targetUserId === targetUserId && check.match);
-};
-
-const isChecked = (userDialogueChecks: DialogueCheckInfo[], targetUserId: string): boolean => {
-  return userDialogueChecks?.find(check => check.targetUserId === targetUserId)?.checked || false;
-};
-
-type MatchDialogueButtonProps = {
-  isMatched: boolean;
-  targetUserId: string;
-  targetUserDisplayName: string;
-  currentUser: UsersCurrent;
-  loadingNewDialogue: boolean;
-  createDialogue: (title: string, participants: string[]) => void;
-  classes: ClassesType;
-};
-
 const MatchDialogueButton: React.FC<MatchDialogueButtonProps> = ({
   isMatched,
+  checkId,
   targetUserId,
   targetUserDisplayName,
   currentUser,
-  loadingNewDialogue,
-  createDialogue,
   classes,
 }) => {
+
+  const { openDialog } = useDialog();
+  const { history } = useNavigation();
+
+  const {loading: userLoading, results} = useMulti({
+    terms: {
+      view: "dialogueMatchPreferences",
+      dialogueCheckId: checkId,
+      limit: 1000,
+    },
+    fragmentName: "DialogueMatchPreferenceInfo",
+    collectionName: "DialogueMatchPreferences",
+  });
+
   if (!isMatched) return <div></div>; // need this instead of null to keep the table columns aligned
 
-  return (
-    <div>
+  const userMatchPreferences = results && results.length > 0 ? results[0] : null;
+  const generatedDialogueId = userMatchPreferences?.generatedDialogueId;
+
+  const renderButton = () => {
+    if (!!generatedDialogueId) {
+      return (
+        <button className={classes.lightGreenButton} onClick={(e) => redirect(generatedDialogueId, history)}>
+          <a data-cy="message">Go to dialogue</a>
+        </button>
+      );
+    }
+  
+    if (userMatchPreferences) {
+      return (
+        <button className={classes.waitingButton} disabled>
+          <a data-cy="message">Waiting for {targetUserDisplayName}...</a>
```

Comment:
This <a> seems unnecessary and probably not good to have. I'd suggest removing

Reply:
which part?

Reply:
(Displaying a different button on userMatchPreferences = true is a desired behavior, but you  might be pointing to something else?)

Reply:
(I think he meant the `<a>` tag; I thought there was another comment somewhere that mentioned that specifically but maybe not)

Reply:
It's gone now!